internal/controller/pipelinerollout_controller.go

```diff
@@ -338,15 +338,25 @@ func applyPipelineSpec(
 	restConfig *rest.Config,
 	obj *kubernetes.GenericObject,
 	pipelineRollout *apiv1.PipelineRollout,
+	rolloutChildOp RolloutChildOperation,
+	childResouceSpecHash string,
 ) error {
 	numaLogger := logger.FromContext(ctx)
-	// TODO: use UpdateSpec instead
-	err := kubernetes.ApplyCRSpec(ctx, restConfig, obj, "pipelines")
-	if err != nil {
-		numaLogger.Errorf(err, "failed to apply Pipeline: %v", err)
-		pipelineRollout.Status.MarkFailed("ApplyPipelineFailure", err.Error())
-		return err
+
+	if rolloutChildOp != RolloutChildNoop {
+		// TODO: use UpdateSpec instead
+		err := kubernetes.ApplyCRSpec(ctx, restConfig, obj, "pipelines")
+		if err != nil {
+			numaLogger.Errorf(err, "failed to apply Pipeline: %v", err)
+			pipelineRollout.Status.MarkFailed("ApplyPipelineFailure", err.Error())
+			return err
+		}
+
+		kubernetes.SetAnnotation(pipelineRollout, apiv1.KeyHash, childResouceSpecHash)
```

Comment:
pause/resume the pipeline spec change will cause the hash to be different right? If so, we need to calculate the new hash afterwards.

Reply:
Only the pipeline resource will be updated and we compare the hash of the pipeline rollout `pipeline` spec. So no need to change as discussed on phone.